src/aap_eda/core/management/commands/purge_log_records.py

```diff
@@ -0,0 +1,139 @@
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import logging
+from datetime import datetime
+from typing import Union
+
+from django.core.management.base import BaseCommand, CommandParser
+from django.db import transaction
+from django.db.models import Q
+
+from aap_eda.core import models
+
+logger = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """Purge the logs from a rulebook process."""
+
+    help = (
+        "Purge log records from rulebook processes. "
+        "If activation ids or names are not specified, "
+        "all log records older than the cutoff date will be purged."
+    )
+
+    def add_arguments(self, parser: CommandParser) -> None:
+        parser.add_argument(
+            "--activation-ids",
+            nargs="+",
+            type=int,
+            dest="activation-ids",
+            help=(
+                "Specify the activation ids which you want to clear their "
+                "records (e.g., ActivationID1 ActivationID2)"
+            ),
+        )
+        parser.add_argument(
+            "--activation-names",
+            nargs="+",
+            type=str,
+            dest="activation-names",
+            help=(
+                "Specify the activation names which you want to clear their "
+                "records (e.g., ActivationName1 ActivationName2)"
+            ),
+        )
+        parser.add_argument(
+            "--date",
+            dest="date",
+            action="store",
+            required=True,
+            help=(
+                "Purge records older than this date from the database. "
+                "The cutoff date in YYYY-MM-DD format"
+            ),
+        )
+
+    def purge_log_records(
+        self, ids: list[int], names: list[str], cutoff_date: str
+    ) -> None:
+        new_instance_logs = []
+        if not bool(ids) and not bool(names):
+            instances = models.RulebookProcess.objects.all()
+        else:
+            instances = models.RulebookProcess.objects.filter(
+                Q(activation__id__in=ids) | Q(activation__name__in=names),
+            )
+
+        if not instances.exists():
+            logger.info("No records has been found for purging.")
+            return
+
+        for instance in instances:
+            new_instance_logs.append(
+                self.clean_instance_logs(instance, cutoff_date)
+            )
+
+        new_instance_logs = [
+            item for item in new_instance_logs if item is not None
+        ]
+        if len(new_instance_logs) > 0:
+            models.RulebookProcessLog.objects.bulk_create(new_instance_logs)
+
+            logger.info(f"Log records older than {cutoff_date} are purged.")
```

Comment:
This should not be a logger:
```suggestion
            self.stdout.write(
                self.style.SUCCESS("Log records older than %s are purged." % cutoff_date)
            )
```